doc/signing-commits.md

```diff
@@ -0,0 +1,61 @@
+# Signing commits
+
+If you are here because you forgot to sign your commits, fear not. Check out [how to sign previous commits](#how-to-sign-previous-commits)
+
+We use developer certificate of origin (DCO) in all hyperledger repositories, so to get your pull requests accepted, you must certify your commits by signing off on each commit. Using a GPG signature is the best way to certify that your commits have come directly from you.
+
+Below we have made a simple walkthrough to do this, with the link to Github's documentation at the bottom.
+
+### Get a GPG key:
+
+1. Check to see if you already have a key: `$ gpg --list-secret-keys --keyid-format LONG`
+
+1. If no key is found, create a new one. If you already have a gpg key, skip to the next step.
+
+  - `$ gpg --gen-key`
+  - Follow the prompts, using the your email that is connected with Github. Using a key size of 4096 is recommended
+  - Once the key's been created, list the new key as in step one
+
+
+## Configure your key with Git:
+  -  The terminal should print something like this:
+
+
+    /Users/hubot/.gnupg/secring.gpg
+    ------------------------------------
+    sec   4096R/<<<<key-id>>>> 2016-03-10 [expires: 2017-03-10]
+    uid                          Hubot
+    ssb   4096R/42B317FD4BA89E7A 2016-03-10
+
+
+  - Export the key using the key-id above: `$ gpg --armor --export <key-id>`
+  - Copy the terminal output beginning with `-----BEGIN PGP PUBLIC KEY BLOCK-----` and ending with `-----END PGP PUBLIC KEY BLOCK-----`.
+  - Add the key to github by navigating to github.com -> settings -> ssh and gpg keys
+  - Add the key to your local installation of git: `git config --global user.signingkey <key-id>`
+  - To add your GPG key to your bash profile, paste the text below:
+      `echo 'export GPG_TTY=$(tty)' >> ~/.bashrc`
+  - [optional, but recommended] To configure your Git client to sign commits by default for  this local repository, run `git config commit.gpgsign true`. If you do this, you no longer need to add the `-S` flag to commits, but you will still need to add the `-s` flag to sign the commit body.
+  - [optional] you can also set up a GPG agent to manage your password so you don't need to enter it in after each commit.
+
+
+## Signing your current commit
+  - `$ git commit -S -s -m "your commit message"`
+  - To see if your commits have been signed off, run `$ git log --show-signature`
+  - If you need to re-sign the most current commit, use `$ git commit --amend --no-edit -S -s`.
+
+The `-S` flag signs your commit with GPG. the `-s` flag signs the commit message with your name and email. Both are required for the DCO check to succeed.
+
+## How to Sign Previous Commits
+   1. Use `git-log --show-signature` to see which commits need to be signed.
```

Comment:
Should this be `git log`? When I use `git-log`, I am told `command not found`

Reply:
@tkuhrt that was a typo. Fixed